### ISSUE TITLE
no-unused-styles doesn't support conditional styles

### ISSUE THREAD

**Issue by robcalcroft** [Intellicode/eslint-plugin-react-native#125]
```javascript
const styles = StyleSheet.create({
  container: {
    padding: 1,
  },
  'container--critical': {
    backgroundColor: MY_RED_CONST,
  },
});

const Component = ({ shouldBeRed }) => (
  <View style={[styles.container].concat(shouldBeRed ? styles['container--critical'] : [])}>
    <Text>This is some text</Text>
  </View>
);
```

**Comment by Intellicode** [comment]
Thanks for reporting, feel free to open a PR or wait until I have time to look into it :)
